explorer/interpreter/builtins.h

```diff
@@ -87,14 +119,45 @@ class Builtins {
     return BuiltinNames[static_cast<int>(builtin)];
   }
 
+  // Get the builtin interface to use for an assignment operator.
+  static Builtin BuiltinInterfaceForAssignOperator(AssignOperator op);
```

Comment:
It doesn't look like other operators are handled this way; wouldn't putting this logic in the type checker be more consistent?

Comparing with:
https://github.com/carbon-language/carbon-lang/blob/trunk/explorer/interpreter/type_checker.cpp#L3102-L3111

Also, if you're doing this, note you should add an explicit include of / dependency on statement.h

Reply:
Moved as suggested.